roles/default/kiali-deploy/tasks/main.yml

```diff
@@ -261,13 +261,13 @@
   when:
   - kiali_vars.external_services.prometheus.url == ""
 
-- name: Set default Prometheus Custom Metrics URL
+- name: "Set default namespace label for custom dashboards: needed for Istio 1.7 and later"
   set_fact:
-    kiali_vars: "{{ kiali_vars | combine({'external_services': {'prometheus': {'custom_metrics_url': kiali_vars.external_services.prometheus.url}}}, recursive=True) }}"
+    kiali_vars: "{{ kiali_vars | combine({'external_services': {'custom_dashboards': {'namespace_label': 'kubernetes_namespace'}}}, recursive=True) }}"
   when:
-  - kiali_vars.external_services.prometheus.custom_metrics_url == ""
+  - ???
```

Comment:
@jmazzitelli any idea?

Reply:
This is a new feature we would need to add. AND we need to make sure this version checking works on both upstream Istio and ServiceMesh.

IIRC, we have already have a fact variable to tell us if we are on maistra (aka SM) versus upstream istio, but right now we have no way of knowing what version of either we are on. And remember, Service Mesh reports different version numbers versus upstream Istio. So this won't be trivial.

Reply:
OK, then I'm just removing this setup, it will have to be manually configured until perhaps we find a solution.